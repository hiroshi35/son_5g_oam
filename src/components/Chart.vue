<template>
  <div>
    <canvas ref="canvas" width="100" :height="height"></canvas>
  </div>
</template>

<script>
  import Chart from 'chart.js'

  export default {
    props: {
      type: {
        default: 'line'
      },
      height: {
        type: Number,
        default: 50
      },
      datasets: {
        required: true
      },
      labels: {
        required: true
      },
      options: {
        default () {
          return {
            maintainAspectRatio: true,
            responsive: true,
            legend: {display: true}
          }
        }
      }
    },
    watch: {
      datasets (val) {
        this.chart.data.datasets = val
        this.chart.update()
      }
    },
    mounted: function () {
      let element = this.$refs.canvas.getContext('2d')
      this.chart = new Chart(element, {
        type: this.type,
        data: {
          labels: this.labels,
          datasets: this.datasets
        },
        options: this.options
      })
    }
  }
</script>
