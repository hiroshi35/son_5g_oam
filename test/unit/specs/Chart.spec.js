import { shallow } from 'vue-test-utils'
import Chart from '@/components/Chart.vue'
import ChartJs from 'chart.js'
const getContext = jest.fn()
window.HTMLCanvasElement.prototype.getContext = getContext
describe('Chart.vue', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(Chart,
      {
        propsData: {
          type: 'line',
          height: 50,
          labels: ['A', 'B', 'C', 'D', 'E'],
          datasets: {
            label: 'My First dataset',
            data: [65, 59, 80, 81, 56]
          }
        }
      })
  })
  it('should be vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
  it('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
  it('has correct props', () => {
    expect(wrapper.props().type).toBe('line')
    expect(wrapper.props().height).toBe(50)
    expect(wrapper.props().labels).toEqual(['A', 'B', 'C', 'D', 'E'])
    expect(wrapper.props().datasets).toEqual({label: 'My First dataset', data: [65, 59, 80, 81, 56]})
  })
  test('chart.js should be called with correct arguments', () => {
    expect(ChartJs).toBeCalled()
    expect(ChartJs.mock.calls[0][1].type).toBe('line')
    expect(ChartJs.mock.calls[0][1].data.labels).toEqual(['A', 'B', 'C', 'D', 'E'])
    expect(ChartJs.mock.calls[0][1].data.datasets.label).toBe('My First dataset');
    expect(ChartJs.mock.calls[0][1].data.datasets.data).toEqual([65, 59, 80, 81, 56]);
  })
  test('el.getContext() should be called', () => {
    expect(getContext).toBeCalled()
  })
  test('labels and datasets are required', () => {
    expect(wrapper.vm.$options.props.labels.required).toBeTruthy()
    expect(wrapper.vm.$options.props.datasets.required).toBeTruthy()
  })
})
