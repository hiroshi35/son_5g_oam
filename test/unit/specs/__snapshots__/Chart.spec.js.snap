// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart.vue has the expected html structure 1`] = `
<div>
  <canvas
    height="50"
    width="100"
  />
</div>
`;
